rust/src/utils.rs

```diff
@@ -44,8 +45,7 @@ fn download_url_to_tmpfile(url: &str) -> io::Result<fs::File> {
 /// Given an input string `s`, replace variables of the form `${foo}` with
 /// values provided in `vars`.  No quoting syntax is available, so it is
 /// not possible to have a literal `${` in the string.
-#[allow(dead_code)]
-pub fn varsubst(instr: &str, vars: &HashMap<String, String>) -> io::Result<String> {
+pub fn varsubst(instr: &str, vars: &HashMap<String, String>) -> Result<String, Error> {
```

Comment:
Minor nit: as this is the first place where this is introduced, for future consistency I'd suggest to write it as either `failure::Fallible<String>` or as `Result<String, failure::Error>` (to avoid confusion with `io::Error`).

Reply:
Ah, yeah I like `Falliable<>`.  Updated, thanks!